Projects/Features/Recommend/Sources/Presentation/RecommendViewModel.swift

```diff
@@ -5,4 +5,177 @@
+      case .farmemeButtonTapped(let memeId):
+        await saveMeme(memeId: memeId)
+      }
+    }
+  }
+}
+
+private extension RecommendViewModel {
+  func getRecommendAndUser() async {
+    do {
+      let recommendMemes = try await getRecommendMemesUseCase.execute(size: 5)
+      let user = try await getUserInfoUseCase.get()
+      
+      self.state.recommendMemes = recommendMemes
+      self.state.userLevel = user.level
+      self.state.memeRecommendWatchCount = user.memeRecommendWatchCount
+      
+    } catch {
+      print("Failed get recommend memes : \(error)")
+    }
+  }
+  
+  func postShownMeme(memeId: String?) async {
+    guard let memeId else { return }
+    do {
+      try await watchMemeUseCase.execute(memeId: memeId, type: "recommend")
+      let user = try await getUserInfoUseCase.get()
+      self.state.userLevel = user.level
+      self.state.memeRecommendWatchCount = user.memeRecommendWatchCount
+    } catch {
+      print("Failed show recommnedMeme : \(error)")
+    }
+  }
+  
+  func postReaction(memeId: String?) async {
+    guard let memeId else { return }
+    do {
+      try await reactToMemeUseCase.execute(memeId: memeId)
+      
+      if let index = self.state.recommendMemes.firstIndex(where: { $0.id == memeId }) {
+        self.state.recommendMemes[index].reaction += 1
+      }
+    } catch {
+      print("Failed post recation : \(error)")
+    }
+  }
+  
+  func copyImage(memeImageUrl: String?) async {
+    guard let memeImageUrl else { return }
+    
+    guard let url = URL(string: memeImageUrl) else { return }
+    
+    do {
+      let (data, _) = try await URLSession.shared.data(from: url)
+      guard let image = UIImage(data: data) else {
+        return
+      }
+
+      UIPasteboard.general.image = image
+    } catch {
+      print("Failed to load image data: \(error)")
+    }
+  }
+  
+  func showShareSheet(memeImageUrl: String?) async {
+    guard let memeImageUrl else { return }
+    
+    guard let url = URL(string: memeImageUrl) else {
+      print("invalid url")
+      return
+    }
+    do {
+      let (data, _) = try await URLSession.shared.data(from: url)
+      guard let image = UIImage(data: data) else {
+        print("invalid image data")
+        return
+      }
+      await self.router?.showShareView(items: [image])
+    } catch {
+      print("Failed to load image data: \(error)")
+    }
+  }
+  
+  func saveMeme(memeId: String?) async {
+    guard let memeId else { return }
+    guard let index = self.state.recommendMemes.firstIndex(where: { $0.id == memeId }) else {
+      print("not found meme. memeId: \(memeId)")
+      return
+    }
+    
+    if self.state.recommendMemes[index].isFarmemed {
+      print("already farmeme.")
+      return
+    }
+    
+    do {
+      try await bookmarkMemeUseCase.execute(memeId: self.state.recommendMemes[index].id)
+      self.state.recommendMemes[index].isFarmemed = true
+    } catch {
+      print("Failed save meme : \(error)")
+    }
+  }
```

Comment:
요기 코드 보니까 index를 사용하는게 아닌 것 같아서 first(where) 활용해서 바꿔도 괜찮을 것 같아요~!
```suggestion
    guard let meme = self.state.recommendMemes.first(where: { $0.id == memeId }) else {
      print("not found meme. memeId: \(memeId)")
      return
    }
    
    if meme.isFarmemed {
      print("already farmeme.")
      return
    }
    
    do {
      try await bookmarkMemeUseCase.execute(memeId: meme.id)
      meme.isFarmemed = true
    } catch {
      print("Failed save meme : \(error)")
    }
  }
```

Reply:
let meme으로 정의하면 안되네요;;
meme 자체도 let 으로 선언되는 순간 하위 프로퍼티가 var이더라도 assign이 안되네용
<img width="1535" alt="image" src="https://github.com/user-attachments/assets/3bf16f5a-fe6f-4681-aea2-51ba64a1924f">

Reply:
`var meme`으로 하면용?_?

Reply:
그래서 var name으로 했어요